src/Mod/Path/PathScripts/PathDeburr.py

```diff
@@ -62,40 +69,52 @@ def toolDepthAndOffset(width, extraDepth, tool):
 
     def initOperation(self, obj):
         PathLog.track(obj.Label)
-        obj.addProperty('App::PropertyDistance', 'Width', 'Deburr', QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'The desired width of the chamfer'))
-        obj.addProperty('App::PropertyDistance', 'ExtraDepth', 'Deburr', QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'The additional depth of the tool path'))
-        obj.addProperty('App::PropertyEnumeration', 'Join', 'Deburr', QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'How to join chamfer segments'))
+        obj.addProperty('App::PropertyDistance', 'Width', 'Deburr',
+                        QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'The desired width of the chamfer'))
+        obj.addProperty('App::PropertyDistance', 'ExtraDepth', 'Deburr',
+                        QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'The additional depth of the tool path'))
+        obj.addProperty('App::PropertyEnumeration', 'Join', 'Deburr',
+                        QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'How to join chamfer segments'))
         obj.Join = ['Round', 'Miter']
         obj.setEditorMode('Join', 2)  # hide for now
-        obj.addProperty('App::PropertyEnumeration', 'Direction',  'Deburr', QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'Direction of Operation'))
+        obj.addProperty('App::PropertyEnumeration', 'Direction', 'Deburr',
+                        QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'Direction of Operation'))
         obj.Direction = ['CW', 'CCW']
-        obj.addProperty('App::PropertyEnumeration', 'Side',  'Deburr', QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'Side of Operation'))
+        obj.addProperty('App::PropertyEnumeration', 'Side', 'Deburr',
+                        QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'Side of Operation'))
         obj.Side = ['Outside', 'Inside']
-        obj.setEditorMode('Side', 2) # Hide property, it's calculated by op
-        obj.addProperty('App::PropertyInteger', 'EntryPoint',  'Deburr', QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'Select the segment, there the operations starts'))
+        obj.setEditorMode('Side', 2)  # Hide property, it's calculated by op
+        obj.addProperty('App::PropertyInteger', 'EntryPoint', 'Deburr',
+                        QtCore.QT_TRANSLATE_NOOP('PathDeburr', 'Select the segment, there the operations starts'))
 
     def opOnDocumentRestored(self, obj):
         obj.setEditorMode('Join', 2)  # hide for now
 
     def opExecute(self, obj):
         PathLog.track(obj.Label)
-        (depth, offset) = toolDepthAndOffset(obj.Width.Value, obj.ExtraDepth.Value, self.tool)
+        try:
+            (depth, offset) = toolDepthAndOffset(obj.Width.Value, obj.ExtraDepth.Value, self.tool)
+        except ValueError as e:
+            msg = "{} \n No path will be generated".format(e)
+            QtGui.QMessageBox.information(None, "Tool Error", msg)
```

Comment:
This makes it impossible to use the op in headless mode, which is the major reason we split all ops into op logic and ...Gui

Reply:
confirmed

Reply:
resolved